ui/src/components/TaskDetailsCard/index.jsx

```diff
@@ -99,45 +105,63 @@ export default class TaskDetailsCard extends Component {
      * A GraphQL task response.
      */
     task: task.isRequired,
-    /**
-     * A collection of GraphQL task responses associated with the given task.
-     */
-    dependentTasks: arrayOf(
-      shape({
-        taskId: string.isRequired,
-        // note that status and metadata may be missing
-        status: shape({
-          state: string,
-        }),
-        metadata: shape({
-          name: string,
-        }),
-      })
-    ),
     dependents: shape({
       edges: arrayOf(task),
       pageInfo,
     }),
     onDependentsPageChange: func.isRequired,
+    user: object,
   };
 
   state = {
     showPayload: false,
+    dependentTasks: null,
+    loading: false,
   };
 
   handleTogglePayload = () => {
     this.setState({ showPayload: !this.state.showPayload });
   };
 
+  componentDidMount() {
+    this.fetchDependentTasks();
+  }
+
+  async fetchDependentTasks() {
+    const { task, user } = this.props;
+
+    this.setState({ loading: true });
+
+    if (task.dependencies.length > 0) {
+      const definitions = await getTaskDefinitions({
```

Comment:
this is also sub-optimal as we are doing two separate calls because we have two methods .. one returns definition, the other status .. when we were adding batch queries we didn't think about adding one that would include everything. 
It doesn't increase complexity, just the size of the response.

Reply:
will address this in https://github.com/taskcluster/taskcluster/issues/7243